src/main/java/uk/gov/hmcts/juror/api/moj/repository/JurorRepositoryImpl.java

```diff
@@ -83,8 +83,16 @@ public JPAQuery<Tuple> fetchFilteredJurorRecords(JurorRecordFilterRequestQuery q
         }
 
         if (null != query.getJurorName()) {
-            partialQuery.where(JUROR.firstName.concat(" ").concat(JUROR.lastName).containsIgnoreCase(
-                query.getJurorName()));
+
+            String[] names = query.getJurorName().split(" ");
+
+            if (names.length == 2) {
+                partialQuery.where(JUROR.firstName.containsIgnoreCase(names[0])
+                    .and(JUROR.lastName.containsIgnoreCase(names[1])));
```

Comment:
some last names have spaces, perhaps we need to see if `names.length >1 `then for each array item in the search criteria, then check whether either first name of last name contains any item in the list?

Reply:
covering the case where we match the first part of the firstname, then the last name can be multipart